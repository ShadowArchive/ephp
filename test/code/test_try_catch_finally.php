<?php

try {
  print "hey!\n";
  throw new Exception("it's broken");
  print "no way!\n";
} catch (Exception $e) {
  print "Exception: " . $e->getMessage() . "\n";
} finally {
  print "and ends first one\n";
}

try {
  print "hey again!\n";
  throw new Exception("it's broken... again");
  print "no way!\n";
} catch (Exception $e) {
  throw new Exception("Another exception in chain!");
} finally {
  print "should this appears?\n";
}
