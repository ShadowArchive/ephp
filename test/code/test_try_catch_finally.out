hey!
Exception: it's broken
and ends first one
hey again!
should this appears?

Fatal error: Uncaught exception 'Exception' with message 'Another exception in chain!' in {{CWD}}/test/code/test_try_catch_finally.php:18
Stack trace:
#0 {main}
  thrown in {{CWD}}/test/code/test_try_catch_finally.php on line 18
